torch/utils/cpp_extension.py

```diff
@@ -1803,15 +1803,34 @@ def _get_cuda_arch_flags(cflags: Optional[List[str]] = None) -> List[str]:
 
         arch_list = _arch_list.split(';')
 
-    flags = []
+    flags: List[str] = []
     for arch in arch_list:
-        if arch not in valid_arch_strings:
-            raise ValueError(f"Unknown CUDA arch ({arch}) or GPU not supported")
-        else:
-            num = arch[0] + arch[2]
-            flags.append(f'-gencode=arch=compute_{num},code=sm_{num}')
-            if arch.endswith('+PTX'):
-                flags.append(f'-gencode=arch=compute_{num},code=compute_{num}')
+        if arch not in valid_known_arch_strings:
+            if arch == "":
+                warnings.warn("Unknown CUDA arch (empty string) requested.")
+                continue
+            elif arch[0].isdigit:
+                # Assumes that a value starting with a number is an actual capability.
+                # If a user specifies a capability like "9.0" or "90", we assume they know
+                # what they're doing and don't want us to break their build.
+                # We can warn them about it, but we shouldn't outright break.
+                warnings.warn(
+                    f"Unknown CUDA arch ({arch}) requested."
+                    " Assuming a newer capability than known was specified and continuing."
+                    " Compilation may fail!"
+                )
+            else:
```

Comment:
I would also consider detecting `-real`/`-virtual` in the suffixes, in which case they should go after `sm_` and `compute_` respectively in `-gencode`. But also I feel like a more reliable approach would have been to offload this logic to something like `FindCUDAToolkit.cmake`